packages/datadog_flutter_plugin/lib/src/tracing/tracing_headers.dart

```diff
@@ -230,7 +234,9 @@ Map<String, String> getTracingHeaders(
             context.spanId.asString(TracingIdRepresentation.decimal);
       }
       headers[DatadogHttpTracingHeaders.origin] = 'rum';
-      headers[DatadogHttpTracingHeaders.samplingPriority] = sampledString;
+      if (shouldInjectHeaders) {
```

Comment:
:blocking

the configuration not only applies to sampling priortiy but also applies to all other types of headers.

In L228, the `context.sampled` should be replaced with `shouldInjectHeaders` ie 

Something like https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift#L78

Here in this case, if customer has set `all` in config and `context.sampled` is false, we are not setting headers.

Reply:
I think this applies other type of headers as well except trace context.

Reply:
👍 this a change in previous functionality then, at least in Flutter. Previously we only wrote the Datadog  headers for the trace id, tags, and parent id when the trace was sampled in, but wrote origin and sampling priority in all cases.

I double check the logic but I think the Datadog headers are the only ones where I set this logic.

Reply:
@ganeshnj running a comparison in the iOS code we have:
  * [Datadog Header](https://github.com/DataDog/dd-sdk-ios/blob/72c95e5edadc6c3bfc2dbaf563f2e92854c7e5cd/DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift#L78) -- `traceId`, `traceIdTag`, `parentId` headers are written regardless of sample state when TraceContextInjection.all (incorrect in current Flutter code, but different from Flutter's previous implementations)
  * [b3](https://github.com/DataDog/dd-sdk-ios/blob/72c95e5edadc6c3bfc2dbaf563f2e92854c7e5cd/DatadogInternal/Sources/NetworkInstrumentation/B3/B3HTTPHeadersWriter.swift#L136) -- unsampled only writes the sampling string "0" when TraceContextInjection.all
  * [b3multi](https://github.com/DataDog/dd-sdk-ios/blob/72c95e5edadc6c3bfc2dbaf563f2e92854c7e5cd/DatadogInternal/Sources/NetworkInstrumentation/B3/B3HTTPHeadersWriter.swift#L125) -- unsampled only writes the `multipleSampled` header when TraceContextInjection.all
  * [tracecontext](https://github.com/DataDog/dd-sdk-ios/blob/72c95e5edadc6c3bfc2dbaf563f2e92854c7e5cd/DatadogInternal/Sources/NetworkInstrumentation/W3C/W3CHTTPHeadersWriter.swift#L90) -- always writes a the full traceparent and trace state when TraceContextInjection.all

So I think everything in Flutter matches iOS except for the Datadog headers.  Can we confirm that that's the expected behavior on Datadog's traces before making the change?